tests/unit/Utilities/001LocalIndexing_UnitTest.cpp

```diff
@@ -0,0 +1,147 @@
+
+    // Validate with empty unknowns
+    Utilities::ElementLocalIndexing indexing(numberOfUnknowns);
+    INFO("Initial index should be empty");
+    REQUIRE(indexing.getIncludedUnknowns().empty() == true);
+    for (const Base::Element* element : mesh.getElementsList()) {
+        indexing.reinit(element);
+        INFO("Element should match the reinit");
+        REQUIRE(indexing.getElement() == element);
+        indexing.validate();
+    }
+
+    // Switch unknowns and revalidate
+    std::vector<std::size_t> unknowns(1, 1);
+    indexing.reinit(unknowns);
+    indexing.validate();
+    for (const Base::Element* element : mesh.getElementsList()) {
+        indexing.reinit(element);
+        indexing.validate();
+    }
+
+    // Verify with local data
+    unknowns.resize(3);
+    unknowns[0] = 0;
+    unknowns[1] = 1;
+    unknowns[2] = 2;
+    indexing.reinit(unknowns);
+    INFO("Included unknowns should match reinit");
+    REQUIRE(indexing.getIncludedUnknowns() == unknowns);
+    for (const Base::Element* element : mesh.getElementsList()) {
+        indexing.reinit(element);
+        INFO("Check total basis function count");
+        REQUIRE(indexing.getNumberOfDoFs() ==
+                element->getTotalNumberOfBasisFunctions());
+        INFO("Offset for first unknown should be zero")
+        REQUIRE(indexing.getDoFOffset(0) == 0);
+        std::size_t dofZeroSize = element->getNumberOfBasisFunctions(0);
+
+        INFO("Correct number of DoFs for unknown 0");
+        REQUIRE(indexing.getNumberOfDoFs(0) == dofZeroSize);
+
+        INFO("Only the first unknown DoFs are before those of the second one");
+        REQUIRE(indexing.getDoFOffset(1) == dofZeroSize);
+        // More can be added if desired
+    }
+
+    // Check with partial unknowns, dropping the first unknown
+    unknowns.resize(2);
+    unknowns[0] = 1;
+    unknowns[1] = 2;
+    indexing.reinit(unknowns);
+    for (const Base::Element* element : mesh.getElementsList()) {
+        indexing.reinit(element);
+        INFO("First included unknown should have offset 0");
+        REQUIRE(indexing.getDoFOffset(1) == 0);
+        std::size_t dof1Size = element->getNumberOfBasisFunctions(1);
+        INFO("Correct number of DoFs for first unknown");
+        REQUIRE(indexing.getNumberOfDoFs(1) == dof1Size);
+        INFO("Correct offset of second unknown");
+        REQUIRE(indexing.getDoFOffset(2) == dof1Size);
+    }
+
+    // Deassociate with an element and check
+    indexing.reinit(nullptr);
+    INFO("Without element there are no DoFs");
+    REQUIRE(indexing.getNumberOfDoFs() == 0);
+}
+
+int main(int argc, char* argv[]) {
+
+    // The mesh reader needs an initialized MPI and possibly more. To achieve
+    // this initialize the system using only the program name, as the arguments
+    // to the program are most probably used to direct Catch.
+    Base::parse_options(1, argv);
+
+    int result = Catch::Session().run(argc, argv);
+
+    return result;
```

Comment:
I do not fully understand why this is necessary.

Reply:
There are some things inside the mesh reader (that I use for creating valid elements), that depend on the processor rank (index) in an MPI context. MPI requires initialization before any function, including the functions that ask for the processor rank.

The mesh probably needs only MPI, but I'm not certain that that is the case. So this is the way to initialize the whole DGMax properly.

Reply:
I find it a bit weird, why not intialized MPI inside the test?

Reply:
Because you only need to do it once. 

So if we later add multiple tests, then which runs first? Or if you specify that it only needs to run one of the two? From that point of view I found it more logical to set it up before anything else.